src/views/Users/ViewUser/index.jsx

```diff
@@ -0,0 +1,394 @@
+
+    if (permission.metadata.isAdditional) {
+      setAdditionalPermissions(additionalPermissions.filter(excludePermission));
+    } else {
+      setPermissions(permissions.filter(excludePermission));
+    }
+  };
+
+  const handleRestrictionChange = (permission, restriction) => chips => {
+    const updateRestrictions = entry => {
+      if (entry.name !== permission.name) {
+        return entry;
+      }
+
+      const result = entry;
+
+      result.options[restriction] = chips;
+
+      return result;
+    };
+
+    return permission.metadata.isAdditional
+      ? setAdditionalPermissions(additionalPermissions.map(updateRestrictions))
+      : setPermissions(permissions.map(updateRestrictions));
+  };
+
+  const handleRestrictionTextChange = (permission, key) => value => {
+    const updateText = entry => {
+      if (entry.name !== permission.name) {
+        return entry;
+      }
+
+      const result = entry;
+
+      result.metadata[key] = value;
+
+      return result;
+    };
+
+    return permission.metadata.isAdditional
+      ? setAdditionalPermissions(additionalPermissions.map(updateText))
+      : setPermissions(permissions.map(updateText));
+  };
+
+  const handleUserSave = () => {};
+  const handleUserDelete = () => {};
+  const renderRole = (role, index) => (
+    <Grid container spacing={2} key={index}>
+      <Grid item xs={11}>
+        <TextField
+          disabled={role.metadata.isAdditional ? false : !isNewUser}
+          onChange={e => handleRoleNameChange(role, index, e.target.value)}
+          value={role.name}
+          fullWidth
+        />
+      </Grid>
+      <Grid item xs={1} className={classes.gridDelete}>
+        <IconButton onClick={() => handleRoleDelete(role, index)}>
+          <DeleteIcon />
+        </IconButton>
+      </Grid>
+    </Grid>
+  );
+  const renderPermission = (permission, index) => (
+    <Grid container spacing={2} key={index}>
+      <Grid item xs={3}>
+        <AutoCompleteText
+          value={defaultToEmptyString(permission.name)}
+          onValueChange={handlePermissionNameChange(permission, index)}
+          getSuggestions={getSuggestions(allPermissions)}
+          label="Name"
+          required
+          disabled={!permission.metadata.isAdditional}
+          inputProps={{
+            autoFocus: true,
+          }}
+        />
+      </Grid>
+      <Grid item xs={4}>
+        <AutoCompleteText
+          multi
+          disabled={!supportsProductRestriction(permission.name)}
+          selectedItems={permission.options.products}
+          onSelectedItemsChange={handleRestrictionChange(
+            permission,
+            'products'
+          )}
+          onValueChange={handleRestrictionTextChange(permission, 'productText')}
+          value={permission.metadata.productText}
+          getSuggestions={getSuggestions(products)}
+          label="Product Restrictions"
+          inputProps={{
+            autoFocus: true,
```

Comment:
The `autoFocus` should only be for the first text field. We add this prop so that when the user first enters the view, the first text field has the keyboard cursor already set allowing a user to start typing right away. 

This comment applies to the other text fields as well in this component.